### PR TITLE
Step by step documentation of new operations #1457

### PR DESCRIPTION
**Pull request by theoreticalbts**
PR for #1457.  This PR is a work in progress, not ready for merging yet.

### CODE REVIEW

**Comment by theoreticalbts** [comment]
Some TODO items:

- Needs to be rebased after #1464 #1469 are merged to `develop`
- Needs to check for non-sufficient funds
- Needs to actually change balances
- Unit test needs to check balance has changed

File: tests/tests/smt_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#include "../common/database_fixture.hpp"
+
+using namespace steemit::chain;
+using namespace steemit::protocol;
+using fc::string;
+
+BOOST_FIXTURE_TEST_SUITE( smt_tests, clean_database_fixture )
+
+BOOST_AUTO_TEST_CASE( elevate_account )
```
Comment (mvandeberg) [inline tests/tests/smt_tests.cpp]: This does not match the test structure for other ops. Each op has a validate, authority, and apply unit test.

Claim reward balance is a good example of how this should be formatted.

[validate](https://github.com/steemit/steem/blob/master/tests/tests/operation_tests.cpp#L5781-L5834)
[authorities](https://github.com/steemit/steem/blob/master/tests/tests/operation_tests.cpp#L5836-L5859)
[apply](https://github.com/steemit/steem/blob/master/tests/tests/operation_tests.cpp#L6044-L6140)

Comment (Kiwonik) [reply tests/tests/smt_tests.cpp]: As this request is a blocker to further SMT development we'd like to take it over and implement it in dedicated `smt-develop` branch, that is already created and has the `1457-step-by-step` merged to. Do you mind us going for it?

File: libraries/protocol/include/steemit/protocol/config.hpp [hunk 2]
```diff
@@ -43,6 +42,8 @@
 #define STEEMIT_OWNER_AUTH_HISTORY_TRACKING_START_BLOCK_NUM 1
 #else // IS LIVE STEEM NETWORK
 
+#define STEEMIT_BLOCKCHAIN_VERSION              ( version(0, 19, 0) )
```
Comment (mvandeberg) [inline libraries/protocol/include/steemit/protocol/config.hpp]: Changes in develop mean that we do not need to have a testnet version and mainnet version that are different.

**Comment by theoreticalbts** [comment]
Fixed the build, it was missing `#include` file and wrong index name

**Pull request merged**